model/common/src/icon4py/model/common/grid/base.py

```diff
@@ -137,7 +141,7 @@ def _get_offset_provider(self, dim, from_dim, to_dim):
             dim, self.connectivities[dim].dtype
         )
         return gtx.NeighborTableOffsetProvider(
-            xp.asarray(self.connectivities[dim]),
+            np.asarray(self.connectivities[dim]),
```

Comment:
Can you remove the `asarray` here all together? Does that work? We should allocate the connectivity arrays already on the right device such that the do not need to be transferred here.